src/timer/TimerProvider.jsx

```diff
@@ -69,22 +70,35 @@ const TimerProvider = ({
     if (secondsLeft <= 0) {
       Emitter.emit(TIMER_REACHED_NULL);
     }
+
     if (!limitReached && secondsLeft < LIMIT) {
-      clearInterval(timer);
```

Comment:
I've moved this to be handled in the interval block, since I find it easier to understand.